src/pageindicator/index.ts

```diff
@@ -0,0 +1,45 @@
+import pageIndicatorStyle from 'inline:./style.css';
+import { html, LitElement, TemplateResult, unsafeCSS } from 'lit';
+import { property } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { map } from 'lit/directives/map.js';
+import { range } from 'lit/directives/range.js';
+
+export class WPageIndicator extends LitElement {
+  static styles = [unsafeCSS(pageIndicatorStyle)];
+
+  /** Currently selected page (1-based index) */
+  @property({ type: Number, attribute: 'selected-page' })
```

Comment:
im thinking maybe reflect: true here to reflect property changes back to the attribute. There's a small perf cost in this but it's minor unless theres frequent updates which is unlikely for this component I think. Doing so would mean that any time a user does:

```js
document.querySelector('w-pageindicator').selectedPage = 2;
```

then the attribute will be updated as well so that:

```html
<w-pageindicator selected-page="2"></w-pageindicator>
```